File: app/src/main/java/com/arash/altafi/lazyload/oldWay/java/api/ApiClient.java
package com.arash.altafi.lazyload.oldWay.java.api;

import retrofit2.Retrofit;
import retrofit2.adapter.rxjava2.RxJava2CallAdapterFactory;
import retrofit2.converter.gson.GsonConverterFactory;

public class ApiClient {

    public static Retrofit retrofit = null;

    public static Retrofit getRetrofit() {
        if (retrofit == null) {
            retrofit = new Retrofit.Builder()
                    .baseUrl("https://arashaltafi.ir/git/android/")
                    .addConverterFactory(GsonConverterFactory.create())
                    .addCallAdapterFactory(RxJava2CallAdapterFactory.create())
                    .build();
        }
        return retrofit;
    }

}

File: app/src/main/java/com/arash/altafi/lazyload/oldWay/kotlin/api/ApiClient.kt
package com.arash.altafi.lazyload.oldWay.kotlin.api

import retrofit2.Retrofit
import retrofit2.adapter.rxjava2.RxJava2CallAdapterFactory
import retrofit2.converter.gson.GsonConverterFactory

class ApiClient {

    companion object {

        var retrofitKotlin: Retrofit? = null

        fun getRetrofit(): Retrofit {
            if (retrofitKotlin == null) {

                retrofitKotlin = Retrofit.Builder()
                    .baseUrl("https://arashaltafi.ir/git/android/")
                    .addConverterFactory(GsonConverterFactory.create())
                    .addCallAdapterFactory(RxJava2CallAdapterFactory.create())
                    .build()
            }

            return retrofitKotlin!!
        }

    }

}